sway/tree/layout.c

```diff
@@ -48,6 +47,19 @@ static void output_layout_change_notify(struct wl_listener *listener,
 	arrange_windows(&root_container, -1, -1);
 }
 
+struct sway_container *container_set_layout(struct sway_container *container,
+		enum sway_container_layout layout) {
+	if (container->type == C_WORKSPACE) {
+		container->workspace_layout = layout;
+		if (layout == L_HORIZ || layout == L_VERT) {
+			container->layout = layout;
```

Comment:
Is there a reason for `workspace_layout` and `layout` to both exist

Reply:
In my testing, they seem to be effective in that workspaces honor workspace_layout only, but I'm  not sure because I haven't looked into that part of the code deeply. Make an issue and I'll look into it.